### PR TITLE
AB#3020 -- Add instrumentation language switcher API

### PR DESCRIPTION
**Pull request by gregory-j-baker**
### Description

Add instrumentation (counters) to language switcher API.

### Related Azure Boards Work Items

- [AB#3020](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3020)

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/routes/api+/switch-language.ts [hunk 1]
```diff
@@ -1,89 +1,63 @@
...
-  logger.warn(`Method Not Allowed: The requested method (GET) is not allowed for this resource. Allowed methods: PUT.`);
-  throw json(
-    {
-      error: 'Method Not Allowed',
-      message: 'The requested method is not allowed for this resource. Allowed methods PUT.',
-    },
-    { status: 405 },
-  );
+export async function loader({ request }: LoaderFunctionArgs) {
+  logger.warn(`Method Not Allowed: The requested method (${request.method}) is not allowed for this resource. Allowed methods: PUT.`);
```
Comment (sebastien-comeau) [inline frontend/app/routes/api+/switch-language.ts:20]: `loader` can only be called by GET method

Comment (gregory-j-baker) [reply frontend/app/routes/api+/switch-language.ts:20]: I know, but I changed the log line to be consistent with the other one in the file for easier maintainability.

**Review by sebastien-comeau — APPROVED** [review]

File: frontend/app/routes/api+/switch-language.ts (hunk 1 above)
Comment (gregory-j-baker) [reply frontend/app/routes/api+/switch-language.ts:20]: Actually.. you forced me to check. (I added the IN LOADER bit to make sure it was the loader that was logging)

```
IN LOADER --- Method Not Allowed: The requested method (OPTIONS) is not allowed for this resource. Allowed methods: PUT.
```

```
IN LOADER --- Method Not Allowed: The requested method (HEAD) is not allowed for this resource. Allowed methods: PUT.
```

Comment (sebastien-comeau) [reply frontend/app/routes/api+/switch-language.ts:20]: Good catch!

**Pull request merged**